base_layer/core/src/validation/mocks.rs

```diff
@@ -108,22 +109,17 @@ impl InternalConsistencyValidator for MockValidator {
 impl<B: BlockchainBackend> HeaderChainLinkedValidator<B> for MockValidator {
     fn validate(
         &self,
-        _: &B,
+        db: &B,
         header: &BlockHeader,
         _: &BlockHeader,
         _: &[EpochTime],
         _: Option<Difficulty>,
     ) -> Result<AchievedTargetDifficulty, ValidationError> {
         if self.is_valid.load(Ordering::SeqCst) {
-            let achieved = sha3x_difficulty(header)?;
-
-            let achieved_target = AchievedTargetDifficulty::try_construct(
-                PowAlgorithm::Sha3x,
-                achieved,
-                achieved.checked_add(1).unwrap(),
-            )
-            .unwrap();
-            Ok(achieved_target)
+            // TODO: this assumes consensus rules are the same as the test rules which is a little brittle
```

Comment:
```suggestion
            // this assumes consensus rules are the same as the test rules which is a little brittle
```

Reply:
I think its a valid assumption. Where this is not the case, we can update the test to reflect this

Reply:
Will remove the TODO